packages/astro/src/core/build/plugins/plugin-pages.ts

```diff
@@ -74,15 +69,6 @@ function vitePluginPages(opts: StaticBuildOptions, internals: BuildInternals): V
 						imports.push(`import { renderers } from "${RENDERERS_MODULE_ID}";`);
 						exports.push(`export { renderers };`);
 
-						// The middleware should not be imported by the pages
-						if (shouldBundleMiddleware(opts.settings)) {
-							const middlewareModule = await this.resolve(MIDDLEWARE_MODULE_ID);
-							if (middlewareModule) {
-								imports.push(`import { onRequest } from "${middlewareModule.id}";`);
-								exports.push(`export { onRequest };`);
-							}
-						}
```

Comment:
Why was this logic removed? A middleware shouldn't be part of an Astro route/page when we are in the edge

Reply:
a middleware is now never part of an Astro route/page

Reply:
Curious why too. Related to edge?

Reply:
To detach middleware from routes. We had to import a route before importing middleware. Now, middleware lives in the manifest instead.

The snippet here serves the purpose of excluding middleware when it is supposed to go into edge middleware. That purpose is served by this line now.

https://github.com/withastro/astro/pull/9776/files#diff-524e25c0bd5feda71ed7489963414a0c1f06da3caa327933574a72b85b668bddR245